tools/eqget/EqBase.py

```diff
@@ -0,0 +1,573 @@
+        GOverR0 = np.array(GOverR0)
+
+        return {
+            'R0': self.R0,
+            'r': radius,
+            'psi': psi,
+            'kappa': kappa,
+            'delta': delta,
+            'Delta': Delta,
+            'GOverR0': GOverR0
+        }
+
+
+    def _get_eq_parameters(self, psi_n):
+        """
+        Calculates the magnetic equilibrium parameters used by the
+        analytical magnetic field in DREAM for a *SINGLE* flux surface.
+        """
+        R, Z = self.get_flux_surface(psi_n)
+
+        rho = self.rho(psi_n)
+        r_minor = rho * self.a_minor
+        Zind = np.argmax(abs(Z))
+
+        R_upper = R[Zind]
+        drho_dpsi = self.rho.derivative()(psi_n)
+        R0 = self.R0
+        R_major = self.R_major(psi_n)
+
+        # Shaping parameters
+        psi = self.psi(R[0], Z[0])[0,0]
+        kappa = (max(Z)-min(Z)) / (2*r_minor)
+        delta = (R_major - R_upper) / r_minor
+        #Delta = self.R_major.derivative()(psi_n) / drho_dpsi / self.a_minor
+        Delta = (max(R)+min(R))/2 - R0
+        GOverR0 = self.f_psi(psi_n) / R0
+
+        return {
+            'r_minor': r_minor,
+            'psi': psi,
+            'kappa': kappa,
+            'delta': delta,
+            'Delta': Delta,
+            'GOverR0': GOverR0,
+            'R': R,
+            'Z': Z
+        }
+
+
+    def get_Bpol_coeff(self):
+        """
+        Return the COCOS conversion coefficient for the poloidal
+        magnetic field.
+        """
+        cocos = self.cocos()
+        coeff = cocos['sigma_RphiZ']*cocos['sigma_Bp']/(2*np.pi)**cocos['exp_Bp']
+        return coeff
+        
+    def get_Br(self, R, Z):
+        """
+        Return the radial magnetic field component on the given (R,Z) grid.
+        """
+        coeff = self.get_Bpol_coeff()
+        Br = coeff/R * self.psi(R, Z, dy=1, grid=False)
+        return Br
+
+
+    def get_Bz(self, R, Z):
+        """
+        Return the vertical magnetic field component on the given (R,Z) grid.
+        """
+        coeff = self.get_Bpol_coeff()
+        Bz = -coeff/R * self.psi(R, Z, dx=1, grid=False)
+        return Bz
+
+
+    def get_Btor(self, R, Z):
+        """
+        Return the toroidal magnetic field component on the given (R,Z) grid.
+        """
+        psi = self.psi(R, Z, grid=False)
+        psi_n = (psi - self.psi_axis) / (self.psi_bdry - self.psi_axis)
+
+        f  = self.f_psi(psi_n)
+        Btor = f / R
+
+        return Btor
+
+
+    def getBOfTheta(self, psi_n):
+        """
+        Evaluates the magnetic field strength along the specified flux
+        surface as a function of poloidal angle.
+        """
+        R, Z = self.get_flux_surface(psi_n)
+
+        Br = self.get_Br(R, Z)
+        Bz = self.get_Bz(R, Z)
+        Bt = self.get_Btor(R, Z)
+
+        B = np.sqrt(Br**2 + Bz**2 + Bt**2)
+        theta = np.arctan2(R-self.R0, Z-self.Z0)
+
+        return theta, B
+
+
+    def load(self, filename, override_psilim=False):
```

Comment:
Indeed :P I think it's a left-over from when the ``GEQDSK`` and ``EqBase`` classes were one and the same. Since ``EqBase`` is supposed to be an abstract base class, it doesn't make sense for it to implement a ``load()`` function. I will push a new version in which it is removed!